### PR TITLE
Allow option to ask Twitter API for the user email address

### PR DESCRIPTION
**Pull request by needcaffeine**

### CODE REVIEW

**Review by na2hiro — CHANGES_REQUESTED** [review]
I think this PR is overall good. However, I cannot successfully get email address in [remix-auth-twitter-example](https://github.com/na2hiro/remix-auth-twitter-example) after publishing this locally. I didn't see any message on email on the Auth page [as the doc mentions](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/manage-account-settings/api-reference/get-account-verify_credentials), so I think I need to change settings of App in Twitter, but I didn't find an option related to emails.

I'll dig into this tonight. Also, have you tried running with your app? If so do you have some advice or caveat?

File: src/index.ts [hunk 1]
```diff
@@ -352,11 +361,13 @@ export class TwitterStrategy<User> extends Strategy<
...
   protected async userProfile(
     accessToken: string,
-    accessTokenSecret: string
+    accessTokenSecret: string,
+    includeEmail: boolean
   ): Promise<TwitterProfile> {
     const params = this.signRequest(
       {
         oauth_token: accessToken,
+        include_email: includeEmail === true ? "true" : "false",
```
[the remark this answers is not in the document]
Comment (needcaffeine) [reply src/index.ts:370]: How would you like me to handle this? I'm effectively converting the boolean to a string.

**Comment by needcaffeine** [comment]
>I think this PR is overall good. However, I cannot successfully get email address in remix-auth-twitter-example after publishing this locally.

@na2hiro You will need to regenerate the user access token according to the [api docs](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/manage-account-settings/api-reference/get-account-verify_credentials). "Please note - Your app will need to regenerate the user access tokens for previously authenticated users to access their email address.

**Comment by na2hiro** [comment]
ok, it was just me not familiar with the new App settings UI and didn't find where are the real settings for these. Found it here.

Screen Shot 2022-02-01 at 19 08 06

I confirmed it's working as expected. Merging now!

**Comment by na2hiro** [comment]
Thanks for the quick resolution!

**Pull request merged**